src/check.c

```diff
@@ -166,10 +270,40 @@ static void tcase_free(TCase * tc)
     check_list_free(tc->ch_sflst);
     check_list_free(tc->unch_tflst);
     check_list_free(tc->ch_tflst);
-
+    tcase_set_tags(tc, NULL);
     free(tc);
 }
 
+unsigned int tcase_matching_tag(TCase *tc, const char *check_for)
+{
+    char *tags;
+    char *tag;
+
+    if ((NULL == tc->tags) || (NULL == check_for))
+    {
+	return 0;
+    }
+    
+    tags = strdup(check_for);
+
+    tag = strtok(tags, " ");
```

Comment:
Is there was a helper method which converted a string into a list of tokens then this parsing would only need to be done once, instead of here and in tcase_set_tags()